Sources/PushNotifications/PushNotifications.swift

```diff
@@ -85,75 +99,354 @@ public struct PushNotifications {
+        guard let url = URL(string: urlString) else {
+            return completion(.error(PushNotificationsError.error("[PushNotifications] - Error while constructing the URL.\nCheck that the URL string is not an empty string or string contains illegal characters.")))
+        }
 
         var mutablePublishRequest = publishRequest
         mutablePublishRequest["interests"] = interests
-
-        urlRequest.httpBody = try? JSONSerialization.data(withJSONObject: mutablePublishRequest)
-        let dataTask = session.dataTask(with: urlRequest) { (data, response, error) in
-            guard
-                let data = data,
-                let httpURLResponse = response as? HTTPURLResponse
-                else {
-                    return // Improve
+        
+        do {
+            let httpBody = try JSONSerialization.data(withJSONObject: mutablePublishRequest)
+            let request = setRequest(url: url, httpMethod: "POST", body: httpBody)
+            
+            networkRequest(request: request) { result in
+                switch result {
+                case .value(let deviceData):
+                    do {
+                        let publishResponse = try JSONDecoder().decode(PublishResponse.self, from: deviceData)
+                        completion(.value(publishResponse.id))
+                    }
+                    catch {
+                        completion(.error(error))
+                    }
+                case .error(let error):
+                    completion(.error(error))
+                }
             }
+        }
+        catch {
+            completion(.error(error))
+        }
+    }
 
-            let statusCode = httpURLResponse.statusCode
-            guard statusCode >= 200 && statusCode < 300, error == nil else {
-                return // Improve
+    /**
+    Publish the given `publishRequest` to the specified users.
+    - Parameter users: Array of strings that contains user ids.
+    - Parameter publishRequest: Dictionary containing the body of the push notification publish request.
+    - Parameter completion: The block to execute when the `publishToUsers` operation is complete.
+    - returns: A non-empty publish id string if successful; or a non-nil `PushNotificationsError` error otherwise.
+    Example usage:
+
+    ````
+    // Pusher Beams Instance Id.
+    let instanceId = "1b880590-6301-4bb5-b34f-45db1c5f5644"
+    // Pusher Beams Secret Key.
+    let secretKey = "F8AC0B756E50DF235F642D6F0DC2CDE0328CD9184B3874C5E91AB2189BB722FE"
+    // PushNotifications instance.
+    let pushNotifications = PushNotifications(instanceId: instanceId, secretKey: secretKey)
+
+    let users = ["user1", "user2", "user3"]
+    // Publish request: APNs, FCM.
+    let publishRequest = [
+        "apns": [
+            "aps": [
+                "alert": "Hello"
+            ]
+        ],
+        "fcm": [
+            "notification": [
+                "title": "Hello",
+				"body":  "Hello, world",
```

Comment:
too much tabbing

Reply:
Fixed in 30490bae731aed993b9437b9774f3097b0c5146e